### PR TITLE
Ex6

### PR DESCRIPTION
**Pull request by muchafm**
Les 10 premiers éléments de la suite de Fibonacci forment l'ensemble {1,2,3,5,8,13,21,34,55,89}. On remarque que chaque élément est constitué de la somme des deux éléments précédents.

On considère une suite de Fibonacci dont l'élément le plus grand est strictement inférieur à 4 millions. TRouver la somme de tous les éléments pairs de cette suite.

### CODE REVIEW

File: ex_06.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @param integer $x 
+ * 
+ * @return array the summary of the Fibonacci suite generate under 4 000 000
```
Comment (jaljo) [inline ex_06.php:6]: I think you should rather pass the limit here, rather than the first term.

File: ex_06.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if($sumValues % 2 == 0) {
+            $sumEvenNumbers += $sumValues;
+        }
+    }
+
+    return $sumEvenNumbers;
+}
+
+$x = 0;
+$sum = generateFibonacciSuite($x);
```
Comment (jaljo) [inline ex_06.php:46]: Its weird to pass 0 as an argument, as the Fibonacci suite will always start by 1, I think you should rather passs the limit under which you want to generate the suit and the compute the sum :thinking:

File: ex_06.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    return $sum;
+}
+
+/**
+ * @param array $sum set of integer
+ * 
+ * @return integer the summary of the even numbers in the array sum
+ */
+function getEvenNumbers(array $sum): int {
+
```
Comment (jaljo) [inline ex_06.php:34]: remove that free line

File: ex_06.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    $sumEvenNumbers = 0;
```
Comment (jaljo) [inline ex_06.php:35]: add an empty line under this one

File: ex_06.php (hunk 2 above)
Comment (jaljo) [reply ex_06.php:46]: There is a naming problem here...
`generateFibonacciSuite ` leads to thinking that this function returns an array of integers, so, its technically not a sum. As you perform the sum in another function `getEvenNumbers`, maybe you can rename this variable to `$suit`

And getEvenNumbers is misleading too. WDYT about `getEvenTermsSum` ?

Comment (muchafm) [reply ex_06.php:46]: getEvenTermsSum is better, I agree. I agree too about changing the variable sum but I just have a stupid question, why "suit" ?

Comment (jaljo) [reply ex_06.php]: "Suite" sorry its a typo :upside_down_face:

Comment (muchafm) [reply ex_06.php]: ok :)

File: ex_06.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * @param integer $limit
+ * 
+ * @return array the summary of the Fibonacci suite generate under the limit
```
Comment (jaljo) [inline ex_06.php:6]: This function does not return the summary, but a set of integers... This comment is missleading IMHO

Comment (muchafm) [reply ex_06.php]: Absolutely ! I'm changing this right now

**Pull request merged**